Fix unlocalized links everywhere

d-alleyne opened the pull request:
#### Changes proposed in this Pull Request

The previous PR, #63116, fixed specific unlocalized URLs in the Domain Management System. This PR fixes issues that were excluded from  #63116 .


#### Testing Instructions

1. Go to https://wordpress.com/domains/manage/ and select a site with a domain.
2. Click on the 3 dots to see the domain settings.
3. Choose from the following options:
- DNS records->Manage
- Transfer
4. Verify that the support URLs on the page have been localized. 


Relates to 381-gh-Automattic/i18n-issues

d-alleyne committed:
Fix unlocalized links everywhere

yuliyan reviewed (changes requested):
I found more files that use links from `/lib/url/support` and are still non-localized:

```
client/components/domains/connect-domain-step/connect-domain-step-support-info-link.jsx
client/components/domains/connect-domain-step/transfer-domain-step-start.tsx
client/components/domains/domain-registration-suggestion/index.jsx
client/components/domains/transfer-domain-step/transfer-restriction-message.jsx
client/components/domains/use-my-domain/utilities/option-info.js
client/components/domains/use-your-domain-step/index.jsx
client/components/gsuite/gsuite-learn-more/index.jsx
client/components/happiness-support/index.jsx
client/components/jetpack/daily-backup-status/status-card/no-backups-yet.jsx
client/components/jetpack/jetpack-disconnected-wpcom/index.tsx
client/components/jetpack/jetpack-disconnected/index.tsx
client/jetpack-cloud/sections/settings/empty-content.tsx
client/lib/domains/resolve-domain-status.tsx
client/lib/jetpack/contact-support-url.js
client/me/concierge/shared/primary-header.js
client/me/purchases/manage-purchase/plan-details/billing-period.jsx
client/me/purchases/manage-purchase/purchase-meta.jsx
client/my-sites/activity/activity-log/threat-alert.jsx
client/my-sites/checkout/cart/cart-messages.tsx
client/my-sites/checkout/checkout-thank-you/domain-registration-details.jsx
client/my-sites/checkout/checkout-thank-you/google-apps-details.jsx
client/my-sites/checkout/composite-checkout/components/additional-terms-of-service-in-cart.tsx
client/my-sites/checkout/composite-checkout/components/domain-refund-policy.jsx
client/my-sites/domains/components/domain-warnings/index.jsx
client/my-sites/domains/domain-management/components/domain/aftermarket-auction-notice.tsx
client/my-sites/domains/domain-management/components/icann-verification/index.jsx
client/my-sites/domains/domain-management/components/transfer-lock-opt-out-form/index.jsx
client/my-sites/domains/domain-management/edit/domain-types/mapped-domain-type.jsx
client/my-sites/domains/domain-management/list/transfer-lock-opt-out.jsx
client/my-sites/domains/domain-management/settings/cards/transferred-domain-details.tsx
client/my-sites/domains/domain-management/transfer/transfer-out/icann-verification.jsx
client/my-sites/domains/domain-management/transfer/transfer-out/transfer-prohibited.jsx
client/my-sites/domains/domain-management/transfer/transfer-out/warning.jsx
client/my-sites/domains/domain-management/transfer/transfer-page/index.tsx
client/my-sites/email/email-providers-comparison/in-depth/data.tsx
client/my-sites/plans/current-plan/jetpack-product-install/index.tsx
client/my-sites/plugins/jetpack-plugins-setup/index.jsx
```

d-alleyne committed:
Fix failing tests by appending a '/' to the path of each URL.

The localizeUrl() function adds a trailing slash '/' to the path in the URL.
This caused tests to fail instances as localized URLs contained a slash
but the original constant in support.js did not.

The trailing slash was difficult to notice when it appeared before a '#'
character.

d-alleyne committed:
Use localizeUrl in locations that were missed

localizeUrl is used in one of the following scenarios:
- where the constant is used in an anchor element
- prior to passing the constant to a React component

If the component uses localizeUrl internally, e.g. PurchaseButton,
then the constant is unaltered.

yuliyan commented on client/my-sites/checkout/cart/cart-messages.tsx:
```diff
@@ -113,7 +114,7 @@ function getInvalidMultisitePurchaseErrorMessage( {
 			{ message }&nbsp;
 			<a
-				href={ JETPACK_SUPPORT + 'backup/#does-jetpack-backup-support-multisite' }
+				href={ localizeUrl( JETPACK_SUPPORT ) + 'backup/#does-jetpack-backup-support-multisite' }
```
Nit: I think the full concatenated URL string should be wrapped in **localizeUrl**, i.e. `localizeUrl( JETPACK_SUPPORT + 'backup/#does-jetpack-backup-support-multisite' )`. Though, the result should be the same.

d-alleyne replied:
Good point

yuliyan commented on client/me/purchases/manage-purchase/plan-details/billing-period.jsx:
```diff
@@ -116,7 +117,8 @@ export class PlanBillingPeriod extends Component {
 										<a
 											href={
-												JETPACK_SUPPORT + 'reconnecting-reinstalling-jetpack/#reconnecting-jetpack'
+												localizeUrl( JETPACK_SUPPORT ) +
```
Nit: I think the full concatenated URL string should be wrapped in **localizeUrl**, i.e. `localizeUrl( JETPACK_SUPPORT + 'reconnecting-reinstalling-jetpack/#reconnecting-jetpack' )`. Though, the result should be the same.

yuliyan commented on client/me/purchases/manage-purchase/purchase-meta.jsx:
```diff
@@ -370,7 +371,8 @@ function RenewErrorMessage( { purchase, translate, site } ) {
 								<a
 									href={
-										JETPACK_SUPPORT + 'reconnecting-reinstalling-jetpack/#reconnecting-jetpack'
+										localizeUrl( JETPACK_SUPPORT ) +
```
Nit: I think the full concatenated URL string should be wrapped in **localizeUrl**, i.e. `localizeUrl( JETPACK_SUPPORT + 'reconnecting-reinstalling-jetpack/#reconnecting-jetpack' )`. Though, the result should be the same.

yuliyan reviewed (approved):
Thank you for taking care of localizing all occurrences of the support URL constants!

I've reviewed the code and left a few comments about some small nits, that don't affect the functionality. So feel free to ignore them, if you decide they shouldn't be changed.

Also, I've tested some of the URLs and confirmed that they are now properly localized.

LGTM!
